### PR TITLE
New storage contracts that makes the Core Vault contract about 20% sm…

### PR DESCRIPTION
**Pull request by crazybuster**
…aller

### CODE REVIEW

**Comment by franckc** [comment]
C'est bien, tu utilises le Français dans tes PRs maintenant. Tu progresses.

File: contracts/contracts/vault/VaultAdmin.sol [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+            Vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function collectRewardTokens(address _strategyAddr) public onlyGovernor {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        strategy.collectRewardToken();
+    }
+
+    /***************************************
+                    Pricing
```
Comment (franckc) [inline contracts/contracts/vault/VaultAdmin.sol:222]: Those pricing methods are not for governor only so why putting them here?

File: contracts/contracts/vault/VaultStorage.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    // Mints over this amount automatically allocate funds. 18 decimals.
+    uint256 public autoAllocateThreshold;
+    // Mints over this amount automatically rebase. 18 decimals.
+    uint256 public rebaseThreshold;
+
+    OUSD oUSD;
+
+    bytes32 constant adminImplPosition = 0xa2bd3d3cf188a41358c8b401076eb59066b09dec5775650c0de4c55187d17bd9;
+
+    //keccak256("OUSD.vault.governor.admin.impl");
```
Comment (franckc) [inline contracts/contracts/vault/VaultStorage.sol:73]: nit: is this a comment for line 71? if yes, put it before that line.

File: contracts/contracts/vault/VaultAdmin.sol (hunk 1 above)
Comment (crazybuster) [reply contracts/contracts/vault/VaultAdmin.sol:222]: To save more space on the Core

Comment (franckc) [reply contracts/contracts/vault/VaultAdmin.sol:222]: Makes sense.

It's kinda weird to call this contract VaultAdmin then if it includes non admin related methods that are called by the DApp... But that's just a naming thing. Maybe we could call it "VaultUtils"? I'm terrible at namig so I let you guys make that very difficult decision :)

**Review by franckc — APPROVED** [review]
LGTM aside from potentially considering renaming VaultAdmin to VaultUtils or equivalent.

**Pull request closed without merging**